### PR TITLE
Added Tests to Dashboard

### PR DESCRIPTION
**Pull request by macintushar**
## Description

This PR adds unit tests for Dashboard components.

image

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would impact existing functionality)
- [ ] Styling change
- [ ] Documentation update
- [ ] Refactoring
- [x] Chore
  
## How Has This Been Tested?
Added Unit Tests and all tests have passed

## Checklist:
- [x] Ensure a branch name is prefixed with `feature`, `bugfix`, `hotfix`, `release`, `style` or `chore` followed by `/` and branch name e.g `feature/add-salesforce-connector`
- [x] Added unit tests for the changes made (if required)
- [x] Have you made sure the commit messages meets the guidelines?
- [x] Added relevant screenshots for the changes
- [x] Have you tested the changes on local/staging?
- [x] Have you made sure the code you have written follows the best practises to the best of your knowledge?

### CODE REVIEW

File: src/views/Dashboard/__tests__/ChartCard.test.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    yDataPoints: ['total_count'],
+    xDataPoints: ['time_slice'],
+    yLabels: { total_count: 'Total' },
+    yData: [
+      { total_count: 30, success_count: 20, failed_count: 10, time_slice: '2022-01-01T00:00:00Z' },
+      { total_count: 25, success_count: 15, failed_count: 10, time_slice: '2022-01-02T00:00:00Z' },
+    ],
+    backgroundColors: { total_count: 'red' },
+  };
+
```
Comment (sumitd94) [inline src/views/Dashboard/__tests__/ChartCard.test.tsx:34]: lets move this into mocks folder

File: src/views/Dashboard/__tests__/ChartCard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  it('renders chart card with correct data', () => {
```
Comment (sumitd94) [inline src/views/Dashboard/__tests__/ChartCard.test.tsx:35]: convent here should be that it starts with `should`

So - `should render chart card ....`

File: src/views/Dashboard/__tests__/ChartCard.test.tsx [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          backgroundColors: {},
+        }}
+        tooltipLabel='Tooltip Label'
+        cardTitle='Card Title'
+        tooltipPosition='top'
+        chartEmptyText='No Data found'
+      />,
+    );
+
+    expect(screen.getByText('No Data found'));
```
Comment (sumitd94) [inline src/views/Dashboard/__tests__/ChartCard.test.tsx:66]: we can cover more of the data here in the test

File: src/views/Dashboard/__tests__/Reports.test.tsx [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+  return {
+    __esModule: true,
+    ChartCard: jest.fn((props) => <div data-testid='mockChartCard' {...props} />),
+  };
+});
+
+it('passes correct props to ChartCard', () => {
+  const mockData: ReportObject[] = [
+    { time_slice: '2021-01-01T00:00:00Z', failed_count: 2, success_count: 98, total_count: 100 },
+  ];
```
Comment (sumitd94) [inline src/views/Dashboard/__tests__/Reports.test.tsx:17]: mocks folder, please

File: src/views/Dashboard/__tests__/ChartCard.test.tsx (hunk 1 above)
Comment (macintushar) [reply src/views/Dashboard/__tests__/ChartCard.test.tsx:24]: Done

File: src/views/Dashboard/__tests__/ChartCard.test.tsx (hunk 2 above)
Comment (macintushar) [reply src/views/Dashboard/__tests__/ChartCard.test.tsx]: Doen

File: src/views/Dashboard/__tests__/Reports.test.tsx (hunk 4 above)
Comment (macintushar) [reply src/views/Dashboard/__tests__/Reports.test.tsx:17]: Added

**Review by sumitd94 — APPROVED** [review]

**Pull request merged**